glean-core/ios/Glean/Glean.swift

```diff
@@ -263,6 +263,64 @@ public class Glean {
         }
     }
 
+    /// When applications are launched using the custom URL scheme, this helper function will process
+    /// the URL and parse the debug commands
+    ///
+    /// - parameters:
+    ///     * url: A `URL` object containing the Glean debug commands as query parameters
+    public func handleCustomUrl(url: URL) {
+        guard url.host?.removingPercentEncoding == "glean",
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let params = components.queryItems else {
+            logger.error("Error parsing query parameters")
+            return
+        }
+
+        var tagName: String?
+        var logPings: Bool?
+        var sendPing: String?
+
+        for param in params {
+            switch param.name {
+            case "tagPings":
+                if let pingTag = param.value,
+                    let pingTagPattern = try? NSRegularExpression(pattern: "^[a-zA-Z0-9-]{1,20}$") {
+                    _ = pingTagPattern.matches(
+                        in: pingTag,
+                        options: [],
+                        range: NSRange(location: 0, length: pingTag.count)
+                    ).map { _ in
+                        tagName = pingTag
+                        logger.debug("Pings tagged with: \(pingTag)")
+                    }
+
+                } else {
+                    logger.error("No value passed to tagPings")
+                }
+            case "logPings":
+                logPings = param.value == "true" ? true : false
+                logger.debug("Log pings set to: \(logPings!)")
+            case "sendPing":
+                sendPing = param.value
```

Comment:
Should we enforce a reasonable value here?
Afaik this just gets passed through, but that means we end up copying it over to Rust and I would hate people passing in very long strings again and again.

Reply:
Sure, we don't enforce any sort of limitations on the Android side of things is the only reason I didn't do any checking here.  It probably shouldn't be too difficult to check this against the collection of registered pings, let me look into it and I'll file a bug to port the same restrictions to Android.

Reply:
Bug for adding the same sort of validation to the `GleanDebugActivity` [filed here](https://bugzilla.mozilla.org/show_bug.cgi?id=1599483)